### PR TITLE
Refs resolver supports text documents

### PR DESCRIPTION
**Pull request by longhotsummer**
Provision refs finders now handle text matches, too.

### CODE REVIEW

**Comment by longhotsummer** [comment]
This is required for the AfricanLII Citator.

File: indigo/analysis/refs/provisions.py [hunk 1]
```diff
@@ -441,13 +506,46 @@ def candidates():
...
 
         return None, None
 
-    def resolve_target(self, node: Element, frbr_uri: str) -> Tuple[Union[str, None], Union[Element, None]]:
+    def get_target_from_text(self, match: re.Match, target: Union[str, None]) -> (Union[str, None], Union[Element, None]):
+        """Determine and resolve a remote target and return the target_frbr_uri and the appropriate root XML element of
+        the document to use to resolve the references. References to the local document are not supported because
+        the local document is assumed to be plain text.
+        """
+        if not target or target == "this":
```
Comment (goose-life) [inline indigo/analysis/refs/provisions.py:520]: do these – 'this', 'thereof', 'of' – mean text ref extraction only works on English documents?

**Review by goose-life — APPROVED** [review]

File: indigo/analysis/refs/provisions.py (hunk 1 above)
Comment (longhotsummer) [reply indigo/analysis/refs/provisions.py:520]: No, they're magic constants that the grammar returns to indicate particular targets types. The grammar's alternative languages can use whatever they need to detect the various conditions, and then return these magic constants.

Comment (goose-life) [reply indigo/analysis/refs/provisions.py:520]: oh amazing! I think I was thrown by them being strings, but that makes sense and is very cool!

**Pull request merged**